app/services/applications_export_for_ucas.rb

```diff
@@ -0,0 +1,86 @@
+# Returns an array of hashes, where each hash is in the format expected by
+# UCAS as 'Dataset 1', defined in the Data Sharing Agreement between us
+class ApplicationsExportForUCAS
+  HEADER_NAMES = {
+    apply_candidate_id: 'Apply candidate ID',
+    first_name: 'First Name',
+    surname: 'Surname',
+    dob: 'DOB',
+    address_line_1: 'Address line 1',
+    address_line_2: 'Address line 2',
+    address_line_3: 'Address line 3',
+    address_line_4: 'Address line 4',
+    country: 'Country',
+    postcode: 'Postcode',
+    email_address: 'Email address',
+    provider_code: 'Provider code',
+    provider_name: 'Provider name',
+    phase: 'Phase',
+    programme_type: 'Programme type',
+    programme_outcome: 'Programme outcome',
+    nctl_subject: 'NCTL subject',
+    course_name: 'Course name',
+    course_code: 'Course Code',
+  }.freeze
+
+  def applications
+    relevant_applications.flat_map do |application_form|
+      application_form.application_choices.map do |choice|
+        convert_to_hash(choice)
+      end
+    end
+  end
+
+  # UCAS require the CSV header to have names which are not very
+  # good as Hash keys, so we convert the first objects' keys to
+  # those human-readable values here
+  def self.csv_header(applications)
+    applications.first.keys.map { |key| HEADER_NAMES[key] }
+  end
+
+private
+
+  def relevant_applications
+    ApplicationForm
+      .includes(
+        :candidate,
+        :application_choices,
+        :application_qualifications,
+        :application_work_experiences,
+        :application_references,
+        application_choices: %i[course provider],
+      )
+      .where('candidates.hide_in_reporting' => false)
+      .where.not(submitted_at: nil)
```

Comment:
I suspect that we only should export applications that have been sent to the provider? You would want to scope to not being in `STATES_NOT_VISIBLE_TO_PROVIDER` instead.

Reply:
Hmm, not sure - all the discussions I've been privy to have been around submitted(-or-further-on) applications. 
I think as the purpose of doing this is to detect duplicates and take preventive action, isn't it the case that if we candetect and correct a duplicate _before_ it goes to the provider, that's actually _better_ than after it's sent?

Reply:
Yeah, that makes sense! Disregard comment.